src/rotary-dialer-element.ts

```diff
@@ -0,0 +1,280 @@
+        );
+        break;
+      case Digit.four:
+        slots?.setAttribute('style', `animation: dialFour ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.five:
+        slots?.setAttribute('style', `animation: dialFive ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.six:
+        slots?.setAttribute('style', `animation: dialSix ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      case Digit.seven:
+        slots?.setAttribute(
+          'style',
+          `animation: dialSeven ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case Digit.eight:
+        slots?.setAttribute(
+          'style',
+          `animation: dialEight ${duration} 1 ${direction} ${timingFn};`
+        );
+        break;
+      case Digit.nine:
+        slots?.setAttribute('style', `animation: dialNine ${duration} 1 ${direction} ${timingFn};`);
+        break;
+      default:
+        slots?.removeAttribute('style');
+    }
+  }
+
+  private static addReverseDialerAnim(slots: SVGAElement, digit: number) {
+    // When the user leaves the mouse (eq of lifting the finger) it goes back
+    RotaryDialerElement.matchDigit(slots, digit, Direction.reverse, '2000ms');
+  }
+
+  private addDialerAnim(digit: number) {
+    this.handleRotation(digit);
+  }
+
+  private removeDialerAnim() {
+    const slots = this.shadowRoot?.querySelector('#slots');
+    slots?.classList.remove('dialer-anim');
+    slots?.removeAttribute('style');
+  }
+
+  private down(digit: number) {
+    this.dial(digit);
+  }
+
+  /**
+   * Exposed lazy dial by applying dial fn with a given arg of num from 0-9
+   * @param digit
+   */
+  public dial(digit: Digit) {
+    this.removeDialerAnim();
+    this.addDialerAnim(digit);
+    this.dispatchEvent(
+      new CustomEvent('dialer-grab', {
+        detail: { digit },
+      })
+    );
+  }
+
+  private onValueChange(event: KeyboardEvent) {
+    const target = event.target as HTMLInputElement;
+    const lastValue = target.value;
+    target.value = '';
+    const digit = parseInt(lastValue);
+    this.dial(digit);
+  }
+
+  private handleRotation(digit: number) {
+    const slots = this.shadowRoot?.querySelector('#slots') as SVGAElement;
+    slots.onanimationend = () => RotaryDialerElement.addReverseDialerAnim(slots, digit);
+    const rAF = requestAnimationFrame(() => {
+      slots?.classList.add('dialer-anim');
+      // When you click on a digit, the circle-hole of that digit
+      // should go all the way until the finger stop.
+      RotaryDialerElement.matchDigit(slots, digit);
+      cancelAnimationFrame(rAF);
+    });
+  }
+
+  render() {
+    return html`
+      <input
+        tabindex="0"
+        type="number"
+        class="hide-input"
+        value="${this.value}"
+        @input="${this.onValueChange}"
+      />
+      <svg width="266" height="266" xmlns="http://www.w3.org/2000/svg">
+        <g fill="none" fill-rule="evenodd">
```

Comment:
I think this outer `g` can be removed (you can move the `fill="none"` up to the `svg` element, if needed

Reply:
removed